packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,395 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { ISharedCell } from "@fluidframework/cell/internal";
+import type { IHostLoader } from "@fluidframework/container-definitions/internal";
+import { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import {
+	CompressionAlgorithms,
+	DefaultSummaryConfiguration,
+} from "@fluidframework/container-runtime/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	IFluidHandle,
+} from "@fluidframework/core-interfaces";
+import type { ISharedMap, ISharedDirectory, SharedDirectory } from "@fluidframework/map/internal";
+import type { SharedString } from "@fluidframework/sequence/internal";
+import {
+	ChannelFactoryRegistry,
+	ITestContainerConfig,
+	DataObjectFactoryType,
+	ITestObjectProvider,
+	createAndAttachContainer,
+	ITestFluidObject,
+	waitForContainerConnection,
+} from "@fluidframework/test-utils/internal";
+
+import type { ISharedMatrix } from "@fluidframework/matrix/internal";
+import type { IConsensusRegisterCollection } from "@fluidframework/register-collection/internal";
+import type { IConsensusOrderedCollection } from "@fluidframework/ordered-collection/internal";
+
+const mapId = "map";
+const stringId = "sharedString";
+const cellId = "cell";
+const counterId = "counter";
+const directoryId = "directory";
+const treeId = "tree";
+const matrixId = "matrix";
+const legacyTreeId = "legacyTree";
+const registerId = "registerCollection";
+const queueId = "consensusQueue";
+const migrationShimId = "migrationShim";
+
+const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+	getRawConfig: (name: string): ConfigTypes => settings[name],
+});
+
+describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) => {
+	const {
+		SharedMap,
+		SharedDirectory,
+		SharedCounter,
+		SharedString,
+		SharedCell,
+		SharedMatrix,
+		ConsensusRegisterCollection,
+		ConsensusQueue,
+	} = apis.dds;
+
+	const registry: ChannelFactoryRegistry = [
+		[mapId, SharedMap.getFactory()],
+		[stringId, SharedString.getFactory()],
+		[cellId, SharedCell.getFactory()],
+		[counterId, SharedCounter.getFactory()],
+		[directoryId, SharedDirectory.getFactory()],
+		// [treeId, SharedTree.getFactory()],
```

Comment:
If the tree implementations are not commented out, I get an error that SharedTree is invalid or not implemented. It might have to do with the fact that SharedTree is not in the test APIs, but I'm not sure yet.